pysmt/simplifier.py

```diff
@@ -280,47 +280,37 @@ def walk_exists(self, formula, args, **kwargs):
         return self.manager.Exists(varset, sf)
 
     def walk_plus(self, formula, args, **kwargs):
-        is_real = any(x.is_real_constant() for x in args)
-        is_int = any(x.is_int_constant() for x in args)
```

Comment:
To check: Do we allow mixing Integer and Real constants in a Plus? I thought we did to avoid the need for explicitly casting the constants, but it is worth checking. If we don't, it seems checking the first argument (as you do below) would be sufficient.

Reply:
I don't think mixed Int - Real constants are supported in math operators +, -, * and /.

If I understand correctly the type checking of math operators is implemented in `type_checker.py` by the method `walk_realint_to_realint`.
The latter calls `walk_type_to_type` with either REAL or INT as arguments.

Also the following simple script throws a PysmtTypeError:
```
from pysmt.shortcuts import *
one = Real(1)
two = Int(2)
three = Plus(one, two)
```